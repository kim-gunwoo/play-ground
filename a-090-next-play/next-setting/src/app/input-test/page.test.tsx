import { fireEvent, render, screen } from '@testing-library/react';
import Page from './page';

describe('/input-test page', () => {
  it('matches snapshot', () => {
    const utils = render(<Page />);
    expect(utils.container).toMatchSnapshot();
  });

  it('input-test page: text fireEvent text', () => {
    render(<Page />);
    const heading = screen.getByRole('heading', { level: 2 });

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('0');

    fireEvent.click(screen.getByRole('button'));

    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent('1');
  });
});
