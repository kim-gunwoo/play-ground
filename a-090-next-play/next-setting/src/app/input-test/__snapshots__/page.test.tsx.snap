// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/input-test page matches snapshot 1`] = `
<div>
  <div>
    <h2>
      0
    </h2>
    <button
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
